mmcls/models/losses/label_smooth_loss.py

```
@@ -1,50 +1,65 @@
 
 @LOSSES.register_module()
-class LabelSmoothLoss(nn.Module):
+class LabelSmoothLoss(CrossEntropyLoss):
+    """Intializer for the label smoothed cross entropy loss.
+        This decreases gap between output scores and encourages generalization.
+        Labels provided to forward can be one-hot like vectors (NxC) or class
+        indices (Nx1).
+        This normalizes the labels to a sum of 1 based on the total count of
+        positive targets for a given sample before applying label smoothing.
+
+        Args:
+            label_smooth_val: value to be added to each target entry
+            reduction: specifies reduction to apply to the output
+        """
 
     def __init__(self,
                  label_smooth_val,
-                 num_classes,
+                 num_classes=None,
                  reduction='mean',
                  loss_weight=1.0):
-        super(LabelSmoothLoss, self).__init__()
-        self.label_smooth_val = label_smooth_val
-        self.avg_smooth_val = self.label_smooth_val / num_classes
-        self.reduction = reduction
-        self.loss_weight = loss_weight
+        super(LabelSmoothLoss, self).__init__(
+            use_sigmoid=False,
+            use_soft=True,
+            reduction=reduction,
+            loss_weight=loss_weight)
+        self._label_smooth_val = label_smooth_val
+        self.num_classes = num_classes
+        self._eps = np.finfo(np.float32).eps
 
-        self.cls_criterion = label_smooth
+    def generate_one_hot_like_label(self, label):
+        """
+        This function takes one-hot or index label vectors and computes
+        one-hot like label vectors (float)
+        """
+        label_shape_list = list(label.size())
+        # check if targets are inputted as class integers
+        if len(label_shape_list) == 1 or (len(label_shape_list) == 2
+                                          and label_shape_list[1] == 1):
+            label = convert_to_one_hot(label.view(-1, 1), self.num_classes)
```

it might be better to throw an exception when label shape is not (C,) or (C,1)